### PR TITLE
feat(http): Improve signal handling and logging

### PR DESCRIPTION
**Pull request by orangejulius**
This PR improves the signal handling of Placeholder so that it exits gracefully in all cases, even when running under Docker.

There appear to have been several interconnected issues:

- SIGINT was not handled, which is the signal sent by `ctrl-C` on the command line and by docker

- the signal handler called `app.close()`, which is not defined. Instead `.close()` must be called on the http server returned from `app.listen()`

Because of the `cluster` module, there's a few different paths through the code: either with multiple CPUs or with a single CPU. In the case of multiple CPUs the signal handler has to work for both a worker and master process.

Along the way, logging has been a bit streamlined and now uses
`pelias-logger`.

### CODE REVIEW

**Comment by orangejulius** [comment]
After submitting this PR I wondered to myself "why does everything work great without defining signal handlers at all with ctrl-c on the command line but not in docker?" and it turns out you can spend a lot of time searching for this answer.

In short, Node.js default signal handlers don't work if Node.js isn't running in its expected environment of a nice cozy standard Linux-ish setup with an init system. When running in Docker, the Node.js process is the only process running in that `PID namespace`, so things don't work as expected.

(read https://github.com/nodejs/node-v0.x-archive/issues/9131#issuecomment-72900581 for more info)

Basically the solutions are:
* use `docker run --init` to wrap the process in a mini init system so Node.js is running in an environment it expects
* manually re-define common signal handlers like `SIGINT`, `SIGTERM`

While it's a bit of a pain, it seems like we might want to define those handlers on our services so that they terminate quickly and cleanly no matter how our users run the Docker containers. So this does indeed look like the right approach.
